tools/commands/build.py

```diff
@@ -66,40 +87,103 @@ def make(self):
+            build_dir = os.path.realpath(modules_dir + '/../multiap/build')
+            install_dir = os.path.realpath(build_dir + '/install')
+
+            if name == 'nng':
+                args.cmake_flags.append("BUILD_SHARED_LIBS=yes")
+                self.builder = cmakebuilder('nng', modules_dir, build_dir, install_dir,
+                        args.native, args.cmake_verbose, args.make_verbose, args.cmake_flags, args.generator)
+
+            if name == 'safeclib':
+                self.builder = acbuilder('safeclib', modules_dir, build_dir, install_dir,
+                        args.host_prefix, args.make_verbose)
+
+            self.run_command(commands)
+
+        # build map modules
+        for name in _map_modules:
+            modules_dir = os.path.realpath(args.map_path)
+            build_dir = os.path.realpath(modules_dir + '/build')
+            install_dir = os.path.realpath(build_dir + '/install')
+
+            self.builder = cmakebuilder(name, modules_dir, build_dir, install_dir,
+                args.native, args.cmake_verbose, args.make_verbose, args.cmake_flags, args.generator)
+
+            self.run_command(commands)
 
-        for name in modules:
-            builder = cmakebuilder(name, modules_dir, build_dir, install_dir, args.native, args.cmake_verbose, args.make_verbose, args.cmake_flags, args.generator)
-            logger.debug(builder)
-            if 'clean' in commands:
-                builder.clean()
-            if 'prepare' in commands:
-                builder.prepare()
-            if 'make' in commands:
-                builder.prepare()
-                builder.make()
+    def run_command(self, commands):
+        logger.debug(self.builder)
+        if 'distclean' in commands:
+            self.builder.distclean()
+        if 'clean' in commands:
+            self.builder.clean()
+        if 'prepare' in commands:
+            self.builder.prepare()
+        if 'make' in commands:
+            self.builder.prepare()
+            self.builder.make()
 
     @staticmethod
     def configure_parser(parser=argparse.ArgumentParser(prog='build')):
         parser.help = "multiap_sw standalone build module"
-        parser.add_argument('modules', choices=['all'] + build_modules, nargs='+', help='module[s] to build')
+        parser.add_argument('modules', choices=['all', 'map', 'dep'] + map_modules + dep_modules, nargs='+', help='module[s] to build')
         parser.add_argument('-c', '--commands', choices=build_targets, nargs='+', default=['make'], help="build command (default is clean+make)")
         parser.add_argument("--verbose", "-v", action="store_true", help="verbosity on")
         parser.add_argument("--native", "-n", action="store_true", help="Build native (not cross compile - ignore external_toolchain.cfg)")
+        parser.add_argument("--host-prefix", "-H", help="cross-compile to build programs to run on HOST (only for building safeclib)")
```

Comment:
Why is this host-prefix needed?

It should actually go the other way round: if `-n` is not given, then the `--host` argument should be given to configure. The value of that argument is unfortunately not so easy to guess. One way would be to extract the `PLATFORM_TOOLCHAIN_PREFIX` from the external_toolchain.cfg and to run `<prefix>-gcc -dumpmachine`.

However, for cross-compilation, you anyway also have to pass `CC=<PLATFORM_TOOLCHAIN_PREFIX>-gcc` and possibly also set CFLAGS.

Therefore, I would leave out the support for cross-compilation of safelibc for the time being.

Reply:
More then that - only map modules should have cross-compilation support, others should simply assume that they are building native always

Reply:
Sounds like it would be best to only compile the map modules unless `-n` is given.